lib/create-contentful-api.ts

```diff
@@ -84,34 +88,46 @@ export type ClientWithoutLinkResolution = BaseClient &
     getEntries<Fields extends FieldsType>(
       query?: EntriesQueries<Fields>
     ): Promise<EntryCollectionWithoutLinkResolution<Fields>>
+
+    parseEntries<Fields extends FieldsType = FieldsType>(
+      data: GenericEntryCollection<Fields>
+    ): EntryCollectionWithoutLinkResolution<Fields>
   }
 
 export type ClientWithAllLocalesAndWithLinkResolutionAndWithUnresolvableLinks = BaseClient &
   BaseClientWithAssetsWithAllLocales & {
     withoutLinkResolution: ClientWithAllLocalesAndWithoutLinkResolution
     withoutUnresolvableLinks: ClientWithAllLocalesAndWithLinkResolutionAndWithoutUnresolvableLinks
-    getEntry<Fields extends FieldsType = FieldsType, Locales extends LocaleCode = any>(
+    getEntry<Fields extends FieldsType = FieldsType, Locales extends LocaleCode = LocaleCode>(
       id: string,
       query?: EntryQueries & { locale?: never }
     ): Promise<EntryWithAllLocalesAndWithLinkResolutionAndWithUnresolvableLinks<Fields, Locales>>
 
-    getEntries<Fields extends FieldsType, Locales extends LocaleCode = any>(
+    getEntries<Fields extends FieldsType, Locales extends LocaleCode = LocaleCode>(
       query?: EntriesQueries<Fields> & { locale?: never }
     ): Promise<
       EntryCollectionWithAllLocalesAndWithLinkResolutionAndWithUnresolvableLinks<Fields, Locales>
     >
+
+    parseEntries<Fields extends FieldsType = FieldsType, Locales extends LocaleCode = LocaleCode>(
+      data: GenericEntryCollectionWithAllLocales<Fields, Locales>
+    ): EntryCollectionWithAllLocalesAndWithLinkResolutionAndWithUnresolvableLinks<Fields, Locales>
   }
 
 export type ClientWithAllLocalesAndWithoutLinkResolution = BaseClient &
   BaseClientWithAssetsWithAllLocales & {
-    getEntry<Fields extends FieldsType, Locales extends LocaleCode = any>(
+    getEntry<Fields extends FieldsType, Locales extends LocaleCode = LocaleCode>(
       id: string,
       query?: EntryQueries & { locale?: never }
     ): Promise<EntryWithAllLocalesAndWithoutLinkResolution<Fields, Locales>>
 
-    getEntries<Fields extends FieldsType, Locales extends LocaleCode = any>(
+    getEntries<Fields extends FieldsType, Locales extends LocaleCode = LocaleCode>(
       query?: EntriesQueries<Fields> & { locale?: never }
     ): Promise<EntryCollectionWithAllLocalesAndWithoutLinkResolution<Fields, Locales>>
+
+    parseEntries<Fields extends FieldsType = FieldsType, Locales extends LocaleCode = LocaleCode>(
+      data: GenericEntryCollectionWithAllLocales<Fields, Locales>
+    ): EntryCollectionWithAllLocalesAndWithoutLinkResolution<Fields, Locales>
```

Comment:
The input and output types are incompatible. If the input data is allowed to have entries, then we can’t disallow them in the output as they will still be there.

Reply:
Good point, we shouldn't allow resolved entries in the input. I changed the types.

Reply:
The changed types allow anything as fields. They’re more generic than the last ones. If you want to disallow resolved entries with and without locales you should use these two:
`NewEntry<Fields, 'WITHOUT_LINK_RESOLUTION'>`
`NewEntry<Fields, 'WITH_ALL_LOCALES' | 'WITHOUT_LINK_RESOLUTION', Locales>`